ui/src/message_popup/playtest/video_alpha_experience_page.jsx

```diff
@@ -0,0 +1,284 @@
+import RecordThenClassifyQuestion from '../linear_session/record_then_classify_question.jsx';
+import {DansonScenarios} from './danson_scenarios.js';
+import type {QuestionT} from './danson_scenarios.js';
+import type {ResponseT} from './danson_scenarios.js';
+import PlainTextQuestion from '../renderers/plain_text_question.jsx';
+import MinimalOpenResponse from '../renderers/minimal_open_response.jsx';
+import VideoRecorder from '../../components/video_recorder.jsx';
+import MinimalTextResponse from '../renderers/minimal_text_response.jsx';
+import MixedQuestion from '../renderers/mixed_question.jsx';
+import ChoiceForBehaviorResponse from '../renderers/choice_for_behavior_response.jsx';
+import * as Routes from '../../routes.js';
+import ReadMore from '../renderers/read_more.jsx';
+
+type NextQuestionT = {
+  id:number,
+  question:QuestionT,
+  responses:[ResponseT]
+};
+
+// The top-level page, manages logistics around email, questions,
+// and the display of instructions, questions, and summary.
+export default React.createClass({
+  displayName: 'VideoAlphaExperiencePage',
+
+  contextTypes: {
+    auth: React.PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    const contextEmail = this.context.auth.userProfile.email;
+    const email = contextEmail === "unknown@mit.edu" ? '' : contextEmail;
+
+    return {
+      email,
+      questions: null,
+      sessionId: uuid.v4()
+    };
+  },
+
+  onStart(email) {
+    const questions = DansonScenarios.questions();
+    this.setState({
+      email,
+      questions
+    });
+  },
+
+  onResetSession() {
+    this.setState(this.getInitialState());
+  },
+
+  onLogMessage(type, response:ResponseT) {
+    const {email, sessionId} = this.state;
+    Api.logEvidence(type, {
+      ...response,
+      sessionId,
+      email,
+      name: email,
+      clientTimestampMs: new Date().getTime(),
+    });
+  },
+
+  onGetNextQuestion(questions:[QuestionT], responses:[ResponseT]) {
+    // This function removes questions that will not be shown to the user based on the user's prior choices.
+    
+    const allQuestions = questions.slice();
+
+    if (responses.length >= allQuestions.length) {
+      return null;
+    }
+    
+    const question = allQuestions[responses.length];
+    return {id: question.id, question:question, responses:responses};
+  },
+
+  render() {
+    return (
+      <SessionFrame onResetSession={this.onResetSession}>
+        {this.renderContent()}
+      </SessionFrame>
+    );
+  },
+
+  renderContent() {
+    const {questions} = this.state;
+    if (!questions) return this.renderIntro();
+
+    return <LinearSession
+      questions={questions}
+      questionEl={this.renderQuestionEl}
+      getNextQuestion={this.onGetNextQuestion}
+      summaryEl={this.renderSummaryEl}
+      onLogMessage={this.onLogMessage}
+    />;
+  },
+
+  renderIntro() {
+    return (
+      <IntroWithEmail defaultEmail={this.state.email} onDone={this.onStart}>
+        <VideoRecorder />
```

Comment:
This isn't where we'd use this normally. Added it here in the intro page just to see whether it works.